typing/src/main/java/org/navgurukul/typingguru/di/TypingModule.kt

```diff
@@ -0,0 +1,15 @@
+package org.navgurukul.typingguru.di
+
+import android.content.SharedPreferences
+import org.koin.dsl.module
+import org.navgurukul.typingguru.utils.TypingGuruPreferenceManager
+import org.navgurukul.typingguru.utils.Utility
+
+private fun provideUtility() : Utility = Utility()
+
+private fun provideTypingGuruPreference(sharedPreferences: SharedPreferences) : TypingGuruPreferenceManager = TypingGuruPreferenceManager(sharedPreferences)
+
+val typingModule = module {
+    single { provideUtility() }
```

Comment:
No real use case to delegate instantiation to a function, objects can be directly instantiated inside the block

Reply:
it is good to have dedicated function for purpose. adding in single line will not make any effect.
But I will make changed as suggested.
val typingModule = module {
    single { Utility() }
    single { TypingGuruPreferenceManager(get()) }
}

Reply:
As a general rule, I try to not encapsulate single line piece of code into a function unless it contains some business logic which is reused at multiple places. Its better to reduce unnecessary function as number of functions are limited in dex file